python/pyspark/sql/utils.py

```diff
@@ -303,3 +304,31 @@ def is_timestamp_ntz_preferred() -> bool:
     """
     jvm = SparkContext._jvm
     return jvm is not None and jvm.PythonSQLUtils.isTimestampNTZPreferred()
+
+
+@contextmanager
+def sql_conf(pairs: Dict[str, Any], *, spark: Optional["SparkSession"] = None) -> Iterator[None]:
```

Comment:
Should probably name it as `sqlConf` since we follow camelcase in the API. In addition, I think this should be able to be imported via something like `from pyspark.sql import sqlConf`. Should also document it in the API reference at `python/docs`